JuicyEngine/Source/Platform/Windows/WindowsWindow.cpp

```diff
@@ -0,0 +1,137 @@
+#include "WindowsWindow.h"
+#include "Core/Log.h"
+#include "Events/ApplicationEvent.h"
+#include "Events/KeyEvent.h"
+#include "Events/MouseEvent.h"
+#include <SDL3/SDL.h>
+
+#include "Platform/Vulkan/VulkanContext.h"
+
+namespace JuicyEngine
+{
+static bool s_SDLInitialized = false;
+
+WindowsWindow::WindowsWindow(const WindowProps& props)
+{
+    Init(props);
+}
+
+WindowsWindow::~WindowsWindow()
+{
+    m_Context->Shutdown();
+}
+
+void WindowsWindow::Init(const WindowProps& props)
+{
+    m_Data.Title = props.Title;
+    m_Data.Width = props.Width;
+    m_Data.Height = props.Height;
+
+    JE_CORE_INFO("Creating window {0} ({1}, {2})", props.Title, props.Width, props.Height);
+
+    if (!s_SDLInitialized)
+    {
+        JE_CORE_ASSERT(SDL_Init(SDL_INIT_VIDEO | SDL_INIT_EVENTS), "Could not initialize SDL3: {0}", SDL_GetError());
+        s_SDLInitialized = true;
+    }
+
+    Uint32 flags = SDL_WINDOW_VULKAN | SDL_WINDOW_RESIZABLE;
+    m_Window = SDL_CreateWindow(m_Data.Title.c_str(), m_Data.Width, m_Data.Height, flags);
+    if (!m_Window)
+    {
+        JE_CORE_ASSERT(m_Window, "Failed to create SDL window: {0}", SDL_GetError());
+    }
+
+    m_Context = new VulkanContext();
+    m_Context->Init();
+    
+    SetVSync(true);
+}
+
+void WindowsWindow::Shutdown()
+{
+    m_Context->Shutdown();
```

Comment:
twice call r m_Context->Shutdown(); (first one in WindowsWindow::~WindowsWindow())